File: octo/msgp/msgp-server/src/test/scala/com/sankuai/octo/scannerReport.scala
package com.sankuai.octo

import dispatch.Defaults._
import dispatch.{as, url, _}
import org.scalatest.{BeforeAndAfter, FunSuite}

class scannerReport extends FunSuite with BeforeAndAfter {
  test("report") {
    while(true) {
      (1 to 340).foreach{
        x =>
          val redirectUri = "http://localhost:8080/api/scanner/report"
          val postReq = url(redirectUri).POST.setContentType("application/json", "UTF-8")
          val a = postReq << """{"appkey":"com.sankuai.octo.scanner","category":"DuplicateRegistry","level":0,"time":1433395169267,"content":"testwyz","identifier":"ip:192.168.0.1|port:8999","roundTime":50,"providerCount":121,"weight":10,"newWeight":30 }"""
          Http(a > as.String)
          Thread.sleep(10)
      }
      println("=========")
      Thread.sleep(5000)
    }
  }
}

File: octo/mns/detector/src/main/java/com/sankuai/octo/model/report/ScannerReport.java
package com.sankuai.octo.model.report;

import com.sankuai.octo.Common;


public class ScannerReport {

    private String appkey = "";
    private String category = "";
    private int level = 0;
    private long time;
    private String content = "";
    private String identifier = "";

    public ScannerReport(String category, String content,
            String identifier) {
        this(0, category, content, identifier);
    }

    public ScannerReport(int level, String category, String content,
            String identifier) {
        this.appkey = Common.appkey;
        this.category = category;
        this.level = level;
        this.time = System.currentTimeMillis();
        this.content = content;
        this.identifier = identifier;
    }


    public String getAppkey() {
        return appkey;
    }

    public void setAppkey(String appkey) {
        this.appkey = appkey;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public int getLevel() {
        return level;
    }

    public void setLevel(int level) {
        this.level = level;
    }

    public long getTime() {
        return time;
    }

    public void setTime(long time) {
        this.time = time;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public String getIdentifier() {
        return identifier;
    }

    public void setIdentifier(String identifier) {
        this.identifier = identifier;
    }
}
